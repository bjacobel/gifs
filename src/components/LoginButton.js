import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';

import { getAuth0AuthAsync } from '../actions/auth';

const mapStateToProps = state => ({
  auth: state.auth,
});

const mapDispatchToProps = {
  getAuth0AuthAsync,
};

class LoginButton extends Component {
  componentWillMount() {
    this.props.getAuth0AuthAsync();
  }

  render() {
    const { auth } = this.props;
    const isAuthenticated = auth.isAuthenticated || false;
    const auth0Service = auth.auth0Service || { login: () => false };

    return (
      <button
        className={ classNames('login-btn', { authed: isAuthenticated }) }
        onClick={ auth0Service.login }
      >
        Sign in
      </button>
    );
  }
}

LoginButton.propTypes = {
  auth: PropTypes.shape({
    isAuthenticated: PropTypes.boolean,
    auth0Service: PropTypes.shape({
      login: PropTypes.func,
    }),
  }),
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(LoginButton);
