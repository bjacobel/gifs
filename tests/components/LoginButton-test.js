import React from 'react';
import { shallow } from 'enzyme';

import LoginButton from '../../src/components/LoginButton';

describe('LoginButton component', () => {
  it('matches snapshot when not authenticated', () => {
    expect(shallow(
      <LoginButton.WrappedComponent
        auth={ {
          isAuthenticated: false,
        } }
        getAuth0AuthAsync={ jest.fn() }
      />,
    )).toMatchSnapshot();
  });

  it('matches snapshot when authenticated', () => {
    expect(shallow(
      <LoginButton.WrappedComponent
        auth={ {
          isAuthenticated: true,
        } }
        getAuth0AuthAsync={ jest.fn() }
      />,
    )).toMatchSnapshot();
  });

  it('calls getAuth0AuthAsync on mount', () => {
    const getAuth0AuthAsync = jest.fn();

    shallow(<LoginButton.WrappedComponent auth={ {} } getAuth0AuthAsync={ getAuth0AuthAsync } />);

    expect(getAuth0AuthAsync).toBeCalled();
  });

  it('calls auth0Service.login on click', () => {
    const login = jest.fn();

    const enzymeRepr = shallow(
      <LoginButton.WrappedComponent
        auth={ {
          isAuthenticated: true,
          auth0Service: { login },
        } }
        getAuth0AuthAsync={ jest.fn() }
      />,
    );

    enzymeRepr.find('button').simulate('click');
    expect(login).toBeCalled();
  });
});
