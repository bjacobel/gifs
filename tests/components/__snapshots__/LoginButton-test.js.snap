exports[`LoginButton component matches snapshot when authenticated 1`] = `
<button
  className="login-btn authed"
  onClick={[Function]}>
  Sign in
</button>
`;

exports[`LoginButton component matches snapshot when not authenticated 1`] = `
<button
  className="login-btn"
  onClick={[Function]}>
  Sign in
</button>
`;
